contracts/value/mod_test.go

```diff
@@ -0,0 +1,235 @@
+	require.EqualError(t, err, fake.Err("failed to DELETE"))
+
+	err = contract.Execute(fakeStore{}, makeStep(t, CmdArg, "DISPLAY"))
+	require.EqualError(t, err, fake.Err("failed to DISPLAY"))
+
+	err = contract.Execute(fakeStore{}, makeStep(t, CmdArg, "fake"))
+	require.EqualError(t, err, "unknown command: fake")
+
+	contract.cmd = fakeCmd{}
+	err = contract.Execute(fakeStore{}, makeStep(t, CmdArg, "SET"))
+	require.NoError(t, err)
+}
+
+func TestCommand_Set(t *testing.T) {
+	contract := NewContract([]byte{}, fakeAccess{})
+
+	cmd := valueCommand{
+		Contract: &contract,
+	}
+
+	err := cmd.set(fake.NewSnapshot(), makeStep(t))
+	require.EqualError(t, err, "'value:key' not found in tx arg")
+
+	err = cmd.set(fake.NewSnapshot(), makeStep(t, KeyArg, "dummy"))
+	require.EqualError(t, err, "'value:value' not found in tx arg")
+
+	err = cmd.set(fake.NewBadSnapshot(), makeStep(t, KeyArg, "dummy", ValueArg, "value"))
+	require.EqualError(t, err, fake.Err("failed to set value"))
+
+	snap := fake.NewSnapshot()
+
+	_, found := contract.index["dummy"]
+	require.False(t, found)
+
+	err = cmd.set(snap, makeStep(t, KeyArg, "dummy", ValueArg, "value"))
+	require.NoError(t, err)
+
+	_, found = contract.index["dummy"]
+	require.True(t, found)
+
+	res, err := snap.Get([]byte("dummy"))
+	require.NoError(t, err)
+	require.Equal(t, "value", string(res))
+}
+
+func TestCommand_Read(t *testing.T) {
+	contract := NewContract([]byte{}, fakeAccess{})
+
+	cmd := valueCommand{
+		Contract: &contract,
+	}
+
+	err := cmd.read(fake.NewSnapshot(), makeStep(t))
+	require.EqualError(t, err, "'value:key' not found in tx arg")
+
+	err = cmd.read(fake.NewBadSnapshot(), makeStep(t, KeyArg, "dummy"))
+	require.EqualError(t, err, fake.Err("failed to get key 'dummy'"))
+
+	snap := fake.NewSnapshot()
+	snap.Set([]byte("dummy"), []byte("value"))
+
+	buf := &bytes.Buffer{}
+	cmd.Contract.printer = buf
+
+	err = cmd.read(snap, makeStep(t, KeyArg, "dummy"))
+	require.NoError(t, err)
+
+	require.Equal(t, "dummy=value", buf.String())
+}
+
+func TestCommand_Delete(t *testing.T) {
+	contract := NewContract([]byte{}, fakeAccess{})
+
+	cmd := valueCommand{
+		Contract: &contract,
+	}
+
+	err := cmd.delete(fake.NewSnapshot(), makeStep(t))
+	require.EqualError(t, err, "'value:key' not found in tx arg")
+
+	err = cmd.delete(fake.NewBadSnapshot(), makeStep(t, KeyArg, "dummy"))
+	require.EqualError(t, err, fake.Err("failed to delete key 'dummy'"))
+
+	snap := fake.NewSnapshot()
+	snap.Set([]byte("dummy"), []byte("value"))
+	contract.index["dummy"] = struct{}{}
+
+	err = cmd.delete(snap, makeStep(t, KeyArg, "dummy"))
+	require.NoError(t, err)
+
+	res, err := snap.Get([]byte("dummy"))
+	require.Nil(t, err)
+	require.Nil(t, res)
+
+	_, found := contract.index["dummy"]
+	require.False(t, found)
+}
+
+func TestCommand_Display(t *testing.T) {
+	contract := NewContract([]byte{}, fakeAccess{})
+	contract.index["key1"] = struct{}{}
+	contract.index["key2"] = struct{}{}
```

Comment:
what guarantees that this would happen on startup?

Reply:
none. This is to manually fill the index, so we can later send the display command.